nixos/modules/services/misc/atuin.nix

```diff
@@ -65,6 +65,7 @@ in
 
     systemd.services.atuin = {
       description = "atuin server";
+      requires = [ "postgresql.service" ];
       after = [ "network.target" "postgresql.service" ];
```

Comment:
Should only be in requires/after, if `ATUIN_DB_URI` is pointing to a database on the same system.

Some part of the change could look like this:
```diff
-      requires = [ "postgresql.service" ];
-      after = [ "network.target" "postgresql.service" ];
+      requires = lib.optionals cfg.database.createLocally [ "postgresql.service" ];
+      after = [ "network.target" ] ++ lib.optionals cfg.database.createLocally [ "postgresql.service" ];
```

Reply:
I've made a small change towards that. It doesn't implement full database configuration, but it does add createLocally. I'm looking to implement the minimal change that fixes the crash on server restart. Hopefully that is enough.